### ISSUE TITLE
logic error in error reporting on upsert failure

### ISSUE THREAD

**Issue by eymiha** [Meteor-Community-Packages/meteor-collection-hooks#316]
in 1.3.2, in os/upsert.js, at line 82,

```
...
const wrappedCallback = function (err, ret) {
  if (err || (ret && ret.insertedId)) {
    // Send any errors to afterInsert
    afterInsert(ret.insertedId, err)
  } else {
...
```

In the function, if err is truthy, ret may still be undefined. Control will flow into the afterInsert, but ret.insertedId will fail to dereference ret and throw it's own error.

At this point, we are blocked as we can't apply the error to the upsert that caused it. So just record it to the server console, I suppose...

```
...
const wrappedCallback = function (err, ret) {
  if (err) {
    if (ret && ret.insertedId) {
      // Send any errors to afterInsert
      afterInsert(ret.insertedId, err)
    } else {
      // unable to recover the id, we need to punt.
      console.error(err);
    }
  } else {
...
```

This error seems to be in all releases of the package. I can only suppose that the problem went unnoticed based on the relative rarity of upserts and the added rarity of problems when upserting.

**Comment by jankapunkt** [comment]
Attempted fix is in tun pipeline: https://github.com/Meteor-Community-Packages/meteor-collection-hooks/pull/319

**Comment by StorytellerCZ** [comment]
@jankapunkt was this fixed?

**Comment by jankapunkt** [comment]
Yes in https://github.com/Meteor-Community-Packages/meteor-collection-hooks/pull/319
